### PR TITLE
Core: Switch tests to Junit5 in avro package

### PR DESCRIPTION
**Pull request by skytin1004**
This pull request is related to the issue "Move JUnit4 tests to JUnit5 https://github.com/apache/iceberg/issues/7160".
Based on the conversations in https://github.com/apache/iceberg/issues/7160 I used AssertJ assertions.

"AvroDataTest","TestAvroDataWriter","TestAvroDeleteWriters","TestAvroEncoderrUitil","TestAvroEnums","TestAvroFileSplit"
,"TestAvroOptionsWithNullDefaults","TestAvroSchemaProjections","TestHasld","TestSchemaConversions"
I migrated these 10 test files from JUnit4 to JUnit5, and they all passed test. and built successfully.

I confirmed that below code already exists in the iceberg-core module part of gradle.build.
```
test {
    useJUnitPlatform()
  }
```

AvroDataTest_passed
TestAvroDataWriter_passed
TestAvroDeleteWriters_passed
TestAvroEncoderUtil_passed
TestAvroEnums_passed
TestAvroFileSplit_passed
TeatAvroOptionsWithNonNullDefaults_passed
TestAvroSchemeaProjection_passed
TestHasIds
TestSchemaConversions

### CODE REVIEW

File: core/src/test/java/org/apache/iceberg/avro/TestAvroDataWriter.java [hunk 1]
```diff
@@ -90,23 +91,24 @@ public void testDataWriter() throws IOException {
...
-    Assert.assertEquals("Record count should match", records.size(), dataFile.recordCount());
-    Assert.assertEquals("Partition should be empty", 0, dataFile.partition().size());
-    Assert.assertEquals(
-        "Sort order should match", sortOrder.orderId(), (int) dataFile.sortOrderId());
-    Assert.assertNull("Key metadata should be null", dataFile.keyMetadata());
+    assertThat(dataFile.format()).as("Format should be Avro").isEqualTo(FileFormat.AVRO);
+    assertThat(dataFile.content()).as("Should be data file").isEqualTo(FileContent.DATA);
+    assertThat(dataFile.recordCount()).as("Record count should match").isEqualTo(records.size());
+    assertThat(dataFile.partition().size()).as("Partition should be empty").isEqualTo(0);
+    assertThat((int) dataFile.sortOrderId())
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/avro/TestAvroDataWriter.java:98]: I think we can remove the casting to `int` here

File: core/src/test/java/org/apache/iceberg/avro/TestAvroDeleteWriters.java [hunk 2]
```diff
@@ -56,9 +56,9 @@ public class TestAvroDeleteWriters {
 
   private List<Record> records;
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir java.nio.file.Path temp;
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/avro/TestAvroDeleteWriters.java:59]: do we need the fully qualified name here? I think we can just add an import here and in the other files

File: core/src/test/java/org/apache/iceberg/avro/TestAvroEnums.java [hunk 3]
```diff
@@ -53,15 +53,15 @@ public void writeAndValidateEnums() throws IOException {
             .endRecord();
 
     org.apache.avro.Schema enumSchema = avroSchema.getField("enumCol").schema().getTypes().get(0);
-    Record enumRecord1 = new GenericData.Record(avroSchema);
+    Record enumRecord1 = new Record(avroSchema);
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/avro/TestAvroEnums.java:56]: are these changes needed?

**Comment by skytin1004** [comment]
@nastra Thank you for your detailed review. I modified the code based on your review.

File: core/src/test/java/org/apache/iceberg/avro/TestAvroEnums.java [hunk 4]
```diff
@@ -71,8 +72,8 @@ public void writeAndValidateEnums() throws IOException {
     }
 
     Schema schema = new Schema(AvroSchemaUtil.convert(avroSchema).asStructType().fields());
-    List<GenericData.Record> rows;
-    try (AvroIterable<GenericData.Record> reader =
+    List<Record> rows;
+    try (AvroIterable<Record> reader =
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/avro/TestAvroEnums.java:76]: nit: I think it would generally be good to not apply changes that aren't related to Junit5 migration

File: core/src/test/java/org/apache/iceberg/avro/TestAvroFileSplit.java [hunk 5]
```diff
@@ -113,16 +112,17 @@ public void testPosField() throws IOException {
...
-          records.get(i).getField(MetadataColumns.ROW_POSITION.name()));
-      Assert.assertEquals(
-          "Field id should match", expected.get(i).getField("id"), records.get(i).getField("id"));
-      Assert.assertEquals(
-          "Field data should match",
-          expected.get(i).getField("data"),
-          records.get(i).getField("data"));
+      assertThat(records.get(i).getField(MetadataColumns.ROW_POSITION.name()))
+          .as("Field _pos should match")
+          .isEqualTo((long) i);
```
[the remark this answers is not in the document]
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/avro/TestAvroFileSplit.java:118]: looks like the casting is still present, do we need it?

File: core/src/test/java/org/apache/iceberg/avro/TestAvroFileSplit.java [hunk 6]
```diff
@@ -136,42 +136,37 @@ public void testPosFieldWithSplits() throws IOException {
...
-          firstHalf.get(i).getField(MetadataColumns.ROW_POSITION.name()));
-      Assert.assertEquals(
-          "Field id should match", expected.get(i).getField("id"), firstHalf.get(i).getField("id"));
-      Assert.assertEquals(
-          "Field data should match",
-          expected.get(i).getField("data"),
-          firstHalf.get(i).getField("data"));
+      assertThat(firstHalf.get(i).getField(MetadataColumns.ROW_POSITION.name()))
+          .as("Field _pos should match")
+          .isEqualTo((long) i);
```
[the remark this answers is not in the document]
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/avro/TestAvroFileSplit.java:152]: casting is still present

File: core/src/test/java/org/apache/iceberg/avro/TestAvroFileSplit.java [hunk 7]
```diff
@@ -136,42 +136,37 @@ public void testPosFieldWithSplits() throws IOException {
...
-          "Field id should match",
-          expected.get(firstHalf.size() + i).getField("id"),
-          secondHalf.get(i).getField("id"));
-      Assert.assertEquals(
-          "Field data should match",
-          expected.get(firstHalf.size() + i).getField("data"),
-          secondHalf.get(i).getField("data"));
+      assertThat(secondHalf.get(i).getField(MetadataColumns.ROW_POSITION.name()))
+          .as("Field _pos should match")
+          .isEqualTo((long) (firstHalf.size() + i));
```
[the remark this answers is not in the document]
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/avro/TestAvroFileSplit.java:164]: casting is still present

**Review by nastra — COMMENTED** [review]
almost ready to go once a few things have been addressed

**Comment by skytin1004** [comment]
@nastra Thank you for your review. I modified the code based on your review :)
 I think the casting should be left as it is, because the test will fail if the units are different.
When I remove the casting, the test failed because the test is comparing an int value to a long value. 
So I think it's correct to cast the i in the long type.

**Review by nastra — APPROVED** [review]
LGTM, thanks @skytin1004
